src/utils/get-types.ts

```diff
@@ -1,9 +1,46 @@
-import { RecordType } from '../index.js'
+import { RecordType, RecordTypeLabel } from '../index.js'
 
-export function getTypes (types?: RecordType | RecordType[]): RecordType[] {
+function recordTypeGuard (type: RecordType | RecordTypeLabel): type is RecordType {
+  return typeof type === 'number' && type in RecordType
+}
+function recordTypeLabelGuard (type: RecordType | RecordTypeLabel): type is RecordTypeLabel {
+  return typeof type === 'string' && type in RecordTypeLabel
+}
+
+export function convertType (type: RecordType | RecordTypeLabel, useRecordTypeValue?: true): RecordType
+export function convertType (type: RecordType | RecordTypeLabel, useRecordTypeValue?: false): RecordTypeLabel
+export function convertType (type: RecordType | RecordTypeLabel, useRecordTypeValue?: boolean): RecordType | RecordTypeLabel
+export function convertType (type: RecordType | RecordTypeLabel, useRecordTypeValue: boolean = true): RecordType | RecordTypeLabel {
+  if (useRecordTypeValue && recordTypeGuard(type)) {
+    return type
+  } else if (!useRecordTypeValue && recordTypeLabelGuard(type)) {
+    return type
+  } else {
+    const reverseMap = {
+      [RecordTypeLabel.A]: RecordType.A,
+      [RecordType.A]: RecordTypeLabel.A,
+      [RecordTypeLabel.CNAME]: RecordType.CNAME,
+      [RecordType.CNAME]: RecordTypeLabel.CNAME,
+      [RecordTypeLabel.TXT]: RecordType.TXT,
+      [RecordType.TXT]: RecordTypeLabel.TXT,
+      [RecordTypeLabel.AAAA]: RecordType.AAAA,
+      [RecordType.AAAA]: RecordTypeLabel.AAAA
+    }
```

Comment:
if this is defined outside of the function, we get `Cannot read properties of undefined (reading 'A')`